src/features/star-pair/store.ts

```diff
@@ -0,0 +1,38 @@
+import { makeAutoObservable } from 'mobx';
+import { Pair, getStarredPairs, setStarredPairs } from './storage';
+
+class StarStateStore {
+  pairs: Pair[] = [];
+
+  constructor() {
+    makeAutoObservable(this);
+
+    if (typeof window !== 'undefined') {
+      this.setup();
+    }
```

Comment:
possible issue: see `connectionStore.setup()` in the `App` component. This was added because this pattern produced hydration errors in the connect store. This may need to go likewise in that useEffect (?).